kura/org.eclipse.kura.core.comm/src/main/java/org/eclipse/kura/core/comm/CommConnectionImpl.java

```diff
@@ -54,42 +56,37 @@
         }
     }
 
-    private final String m_port;
-    private final int m_baudRate;
-    private final int m_dataBits;
-    private final int m_stopBits;
-    private final int m_parity;
-    private final int m_flowControl;
-    private final int m_timeout;
 
-    private final CommURI m_commUri;
-    private SerialPort m_serialPort;
-    private InputStream m_inputStream;
-    private OutputStream m_outputStream;
+    private final CommURI commUri;
+    private SerialPort serialPort;
+    private InputStream inputStream;
+    private OutputStream outputStream;
 
     public CommConnectionImpl(CommURI commUri, int mode, boolean timeouts)
             throws IOException, NoSuchPortException, PortInUseException {
-        this.m_commUri = commUri;
-        this.m_serialPort = null;
-        this.m_port = this.m_commUri.getPort();
-        this.m_baudRate = this.m_commUri.getBaudRate();
-        this.m_dataBits = this.m_commUri.getDataBits();
-        this.m_stopBits = this.m_commUri.getStopBits();
-        this.m_parity = this.m_commUri.getParity();
-        this.m_flowControl = this.m_commUri.getFlowControl();
-        this.m_timeout = this.m_commUri.getTimeout();
+        this.commUri = commUri;
+        this.serialPort = null;
+        
+        final String port = this.commUri.getPort();
+        final int baudRate = this.commUri.getBaudRate();
+        final int dataBits = this.commUri.getDataBits();
+        final int stopBits = this.commUri.getStopBits();
+        final int parity = this.commUri.getParity();
+        final int flowControl = this.commUri.getFlowControl();
+        final int timeout = this.commUri.getTimeout();
 
-        CommPortIdentifier commPortIdentifier = CommPortIdentifier.getPortIdentifier(this.m_port);
+        final CommPortIdentifier commPortIdentifier = CommPortIdentifier.getPortIdentifier(port);
 
-        CommPort commPort = commPortIdentifier.open(this.getClass().getName(), this.m_timeout);
+        final CommPort commPort = commPortIdentifier.open(this.getClass().getName(), timeout);
 
         if (commPort instanceof SerialPort) {
-            this.m_serialPort = (SerialPort) commPort;
+            this.serialPort = (SerialPort) commPort;
             try {
-                this.m_serialPort.setSerialPortParams(this.m_baudRate, this.m_dataBits, this.m_stopBits, this.m_parity);
-                this.m_serialPort.setFlowControlMode(this.m_flowControl);
+                this.serialPort.setSerialPortParams(baudRate, dataBits, stopBits, parity);
+                this.serialPort.setFlowControlMode(flowControl);
             } catch (UnsupportedCommOperationException e) {
-                e.printStackTrace();
+                logger.error("Failed to configure COM port", e);
+                // TODO shouldn't we throw an IOException here
             }
         } else {
             throw new IOException("Unsupported Port Type");
```

Comment:
My personal preference is to use only the KuraException as a checked exception and KuraRuntimeException as an unchecked one. Whenever I need to throw any checked exception, I wrap it in KuraException and throw it. The same for unchecked exception (wrap in KuraRuntimeException).